### ISSUE TITLE
Read the test results out of JUnit reports instad of SBT output as it's hard to parse correctly

### ISSUE THREAD

**Issue by olisikh** [olisikh/neotest-scala#7]
Since all of the libraries generate JUnit XML reports they can be used to get the test results more reliably instead of trying to parse SBT output that may be polluted by logs or other tests that run in parallel.

For example a test file like `com.example.MySpec` would generate a test report at `target/test-reports/TEST-com.example.MySpec` where the report would contain summary and all test results.

When neotest asks to render the test results what the code needs to do:
1. Group the identified tests by path (to avoid doing too much IO / optimize file reading)
2. By each path file the matching JUnit test report
3. Collect the failures per each test (parse it or grep it?)
4. Normalize the failure message text (if needed)
5. Identify the line number of where the failure happened (nice to have, each test lib writes the line number differently, JUnit report may not have the line number unfortunately)
6. Make the neotest test output results according to the interface

**Comment by olisikh** [comment]
Done
